File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/FlipHorizontalTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class FlipHorizontalTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		final float rotation = 180F * position;
		view.setAlpha(rotation > 90F || rotation < -90F ? 0F : 1F);
		view.setPivotX(view.getWidth() * 0.5F);
		view.setPivotY(view.getHeight() * 0.5F);
		view.setRotationY(rotation);
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/FlipHorizontalTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class FlipHorizontalTransformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        val rotation = 1.toFloat() * position
        page.alpha = (rotation > 1.toFloat() || rotation < (-1).toFloat()) then 1.toFloat() ?: 1.toFloat()
        page.pivotX = page.width * 0.1.toFloat()
        page.pivotY = page.height * 0.1.toFloat()
        page.rotationY = rotation
    }

}
